tests/rados/test_node_drain_customer_bug.py

```diff
@@ -0,0 +1,206 @@
+"""
+
+import random
+import re
+import time
+from threading import Thread
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.mgr_workflows import MgrWorkflows
+from ceph.rados.serviceability_workflows import ServiceabilityMethods
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83593996
+    Bug id - https://bugzilla.redhat.com/show_bug.cgi?id=2305677
+    1. Configure a cluster that have more than four OSD nodes
+    2. Select an OSD node and drain the node
+    3. Parallely execute the ceph mgr fail command
+    4. Check that exception occurs on the cluster
+    5. Perform the following workaround  steps-
+          5.1 ceph config-key rm mgr/cephadm/osd_remove_queue
+          5.2 ceph mgr fail
+    6. If exception not occured then check for the Traceback logs
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mgr_obj = MgrWorkflows(node=cephadm)
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    service_obj = ServiceabilityMethods(cluster=ceph_cluster, **config)
+    ceph_nodes = kw.get("ceph_nodes")
+
+    mgr_daemon = Thread(
+        target=background_mgr_task, kwargs={"mgr_object": mgr_obj}, daemon=True
+    )
+    osd_list = []
+
+    for node in ceph_nodes:
+        cmd_host_chk = f"ceph orch host ls --host_pattern {node.hostname}"
+        out = rados_obj.run_ceph_command(cmd=cmd_host_chk)
+        if not out:
+            log.info(f"The {node.hostname} is not in the cluster")
+            continue
+        if node.role == "osd":
+            node_osds = rados_obj.collect_osd_daemon_ids(node)
+            osd_list = osd_list + node_osds
+    osd_weight_chk = check_set_reweight(rados_obj, osd_list)
+    if not osd_weight_chk:
+        log.error(
+            "The osd weights are zero for ew nodes and weights of the OSD are not unique.Set the weights "
+            "manually and re-run the tests"
+        )
+        return 1
+
+    log_lines = [
+        "mgr load Traceback",
+        "TypeError: __init__() got an unexpected keyword argument 'original_weight'",
+    ]
+    ceph_version = rados_obj.run_ceph_command(cmd="ceph version")
+    log.info(f"Current version on the cluster : {ceph_version}")
+    match_str = re.match(
+        r"ceph version (\d+)\.(\d+)\.(\d+)-(\d+)", ceph_version["version"]
+    )
+    major, minor, patch, build = match_str.groups()
+    bug_exists = False
+    exceptional_flag = False
+    if int(major) < 18:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) < 2:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) < 1:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) == 1 and int(build) <= 194:
```

Comment:
Please change the comparison to build # just before the fixed build

Reply:
I changed the comparison  according to the fixed build.